src/tools/cephfs/cephfs-shell

```diff
@@ -1512,37 +1515,61 @@ def get_bool_vals_for_boolopts(val):
-                                                      conf_get('allow_ansi'))
-        shell.echo = get_bool_vals_for_boolopts(cephfs.conf_get('echo'))
-        shell.continuation_prompt = get_bool_vals_for_boolopts(cephfs.
-                                                               conf_get('continuation_prompt'))
-        shell.colors = get_bool_vals_for_boolopts(cephfs.conf_get('colors'))
-        shell.editor = get_bool_vals_for_boolopts(cephfs.conf_get('editor'))
-        shell.feedback_to_output = get_bool_vals_for_boolopts(cephfs.
-                                                              conf_get('feedback_to_output'))
-        shell.max_completion_items = get_bool_vals_for_boolopts(cephfs.
-                                                                conf_get('max_completion_items'))
-        shell.prompt = get_bool_vals_for_boolopts(cephfs.conf_get('prompt'))
-        shell.quiet = get_bool_vals_for_boolopts(cephfs.conf_get('quiet'))
-        shell.timing = get_bool_vals_for_boolopts(cephfs.conf_get('timing'))
-    except (OSError, libcephfs.Error) as e:
-        perror(e)
-        shell.exit_code = e.get_error_code()
+def read_shell_conf(shell, shell_conf_file):
+    import configparser
 
+    sec = 'cephfs-shell'
+    opts = []
+    if LooseVersion(cmd2_version) >= LooseVersion("0.10.0"):
+        for attr in shell.settables.keys():
+            opts.append(attr)
+    else:
+        if LooseVersion(cmd2_version) <= LooseVersion("0.9.13"):
+            # hardcoding options for 0.7.9 because -
+            # 1. we use cmd2 v0.7.9 with teuthology and
+            # 2. there's no way distinguish between a shell setting and shell
+            #    object attribute until v0.10.0
+            opts = ['abbrev', 'autorun_on_edit', 'colors',
+                    'continuation_prompt', 'debug', 'echo', 'editor',
+                    'feedback_to_output', 'locals_in_py', 'prompt', 'quiet',
+                    'timing']
+        elif LooseVersion(cmd2_version) >= LooseVersion("0.9.23"):
+            opts.append('allow_style')
+        # no equivalent option was defined by cmd2.
+        else:
+            pass
 
-if __name__ == '__main__':
-    config_file = ''
+    # default and only section in our conf file.
+    cp = configparser.ConfigParser(default_section=sec, strict=False)
+    cp.read(shell_conf_file)
+    for opt in opts:
+        if cp.has_option(sec, opt):
+            setattr(shell, opt, str_to_bool(cp.get(sec, opt)))
 
+
+def get_shell_conffile_path(arg_conf=''):
+    conf_filename = 'cephfs-shell.conf'
+    env_var = 'CEPHFS_SHELL_CONF'
+
+    arg_conf = '' if not arg_conf else arg_conf
+    home_dir_conf = os.path.expanduser('~/.' + conf_filename)
+    cwd_conf = './' + conf_filename
+    env_conf = os.environ[env_var] if env_var in os.environ else ''
+
+    # here's the priority by which conf gets read.
+    for path in (arg_conf, env_conf, cwd_conf, home_dir_conf):
```

Comment:
let's get rid of `cwd_conf`, that's unexpected for most unix applications.

Reply:
Ok. Fixed.